### PR TITLE
Add the dark version of the IQX color scheme

### PR DESCRIPTION
**Pull request by Cryoris**
### Summary

Add the dark version of the IQX color scheme, as we don't yet support any "dark mode" circuit visualizations which would be very handy e.g. for dark presentations or similar.

### Details and comments

The color codes are taken from IQX and it looks like
image

### CODE REVIEW

**Comment by Cryoris** [comment]
I adapted the background color to the proper IQX one (`#262626`) and I think the PR is ready 👍🏻

**Review by javabster — APPROVED** [review]
LGTM 🚀

**Review by javabster — COMMENTED** [review]
Actually I think you'll need to add new reference images for the snapshot tests, to check 
you can run: https://mybinder.org/v2/gh/cryoris/qiskit-terra/iqx-dark?urlpath=apps/test/ipynb/mpl_tester.ipynb

File: test/ipynb/mpl/circuit/test_circuit_matplotlib_drawer.py [hunk 1]
```diff
@@ -618,15 +620,18 @@ def test_user_style(self):
...
+        for name in ["iqx", "iqx-dark"]:
+            with self.subTest(name=name):
+                circuit = QuantumCircuit(3)
+                circuit.h(0)
+                circuit.x(0)
+                circuit.u(pi / 2, pi / 2, pi / 2, 1)
+                circuit.p(pi / 2, 2)
+                style = {"name": name, "subfontsize": 11}
+
+                self.circuit_drawer(circuit, style=style, filename="subfont.png")
```
Comment (javabster) [inline test/ipynb/mpl/circuit/test_circuit_matplotlib_drawer.py:632]: This causes a naming clash as you're generating 'subfont.png' twice which causes the snapshot test to fail. I don't think it's strictly necessary to test the subfontsize with the new dark theme anyway but i guess it doesn't hurt

```suggestion
                self.circuit_drawer(circuit, style=style, filename="f"{name}_subfont.png"")
```

**Review by javabster — CHANGES_REQUESTED** [review]

**Review by javabster — APPROVED** [review]
Snapshot tests in binder all pass now, thanks @Cryoris 😄

**Review by jakelishman — CHANGES_REQUESTED** [review]
I can't comment on the contrast/accessibility concerns because I really don't know much about them.  Did we decide that it was ok to maintain the white "+" in the cx gates in this theme, in contrast to the general IQX dark theme?

I also think we need to update the test in `qiskit/visualization/matplotlib.py:810` to include `"iqx-dark"`.  In the current form, this code:
```python
from qiskit.circuit import QuantumCircuit
from qiskit.circuit.library import SwapGate
from matplotlib import pyplot
circuit = QuantumCircuit(5)
circuit.append(SwapGate(), [3, 4], [])
circuit.append(SwapGate().control(3), [0, 1, 2, 3, 4], [])
circuit.draw("mpl", style="iqx-dark")
pyplot.show()
```
produces this image:

Figure_1

but the multi-control swap gate should be blue as well.

**Comment by Cryoris** [comment]
Thanks for catch @jakelishman! Also, I adjusted the color of the CX cross to be the same as the background:
image

The contrast question should be solved since we now have the same as IQX, so we're deferring to them 😉

**Review by jakelishman — APPROVED** [review]
Ace, thanks!

**Pull request merged**